### PR TITLE
Enable checks of CM sources compilation by INTEL::online_compiler

### PR DESCRIPTION
**Pull request by v-klochkov**

### CODE REVIEW

File: SYCL/OnlineCompiler/online_compiler_L0.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// REQUIRES: level_zero, level_zero_headers
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -I %level_zero_include_dir -DRUN_KERNELS -lze_loader %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -I %level_zero_include_dir -lze_loader %s -o %th.out
+// RUN: %RUN_ON_HOST %th.out
```
Comment (vladimirlaz) [inline SYCL/OnlineCompiler/online_compiler_L0.cpp]:
```suggestion
// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -DRUN_KERNELS -lze_loader %s -o %t.out
// RUN: %CPU_RUN_PLACEHOLDER %t.out
// RUN: %GPU_RUN_PLACEHOLDER %t.out
// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -lze_loader %s -o %th.out
// RUN: %HOST_RUN_PLACEHOLDER %th.out
```

File: SYCL/OnlineCompiler/online_compiler_OpenCL.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// REQUIRES: opencl
```
Comment (vladimirlaz) [inline SYCL/OnlineCompiler/online_compiler_OpenCL.cpp]:
```suggestion
// REQUIRES: opencl, opencl_icd
```

File: SYCL/OnlineCompiler/online_compiler_OpenCL.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -DRUN_KERNELS -lOpenCL -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -lOpenCL -o %th.out
```
Comment (vladimirlaz) [inline SYCL/OnlineCompiler/online_compiler_OpenCL.cpp]:
```suggestion
// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s %opencl_lib -o %th.out
```

File: SYCL/OnlineCompiler/online_compiler_OpenCL.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+// RUN: %RUN_ON_HOST %th.out
```
Comment (vladimirlaz) [inline SYCL/OnlineCompiler/online_compiler_OpenCL.cpp]:
```suggestion
// RUN: %HOST_RUN_PLACEHOLDER %th.out
```

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: SYCL/OnlineCompiler/online_compiler_L0.cpp [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
```
Comment (vladimirlaz) [inline SYCL/OnlineCompiler/online_compiler_L0.cpp:9]:
```suggestion
// UNSUPPORTED: cl_options

```

File: SYCL/OnlineCompiler/online_compiler_L0.cpp (hunk 1 above)
Comment (v-klochkov) [reply SYCL/OnlineCompiler/online_compiler_L0.cpp]: Thank you. Fixed in a857f2c3e3ab1692e991dba0a5be5bb4ae753181

File: SYCL/OnlineCompiler/online_compiler_OpenCL.cpp (hunk 2 above)
Comment (v-klochkov) [reply SYCL/OnlineCompiler/online_compiler_OpenCL.cpp]: Fixed in a857f2c3e3ab1692e991dba0a5be5bb4ae753181

File: SYCL/OnlineCompiler/online_compiler_OpenCL.cpp (hunk 3 above)
Comment (v-klochkov) [reply SYCL/OnlineCompiler/online_compiler_OpenCL.cpp]: Fixed in a857f2c3e3ab1692e991dba0a5be5bb4ae753181

File: SYCL/OnlineCompiler/online_compiler_OpenCL.cpp (hunk 4 above)
Comment (v-klochkov) [reply SYCL/OnlineCompiler/online_compiler_OpenCL.cpp]: Fixed in a857f2c3e3ab1692e991dba0a5be5bb4ae753181

File: SYCL/OnlineCompiler/online_compiler_L0.cpp (hunk 5 above)
Comment (v-klochkov) [reply SYCL/OnlineCompiler/online_compiler_L0.cpp:9]: Fixed in a857f2c3e3ab1692e991dba0a5be5bb4ae753181

**Review by vladimirlaz — APPROVED** [review]

**Comment by v-klochkov** [comment]
The newly added LIT test passed.
CI on Linux reported a fail for the unrelated test SYCL::InlineAsm/asm_if.cpp

**Pull request merged**